scripts/zones/Promyvion-Mea/npcs/qm1.lua

```diff
@@ -0,0 +1,27 @@
+-----------------------------------
+-- Area: Promyvion Mea
+-- ??? map acquisition
+-----------------------------------
+local ID = require("scripts/zones/Promyvion-Mea/IDs")
+require("scripts/globals/keyitems")
+require("scripts/globals/npc_util")
+-----------------------------------
+
+function onTrigger(player, npc)
+    player:messageSpecial(ID.text.NOTHING_OUT_OF_ORDINARY)
+end
+
+function onTrade(player, npc, trade)
+    if npcUtil.tradeHas(trade, 1722) and not player:hasKeyItem(tpz.ki.MAP_OF_PROMYVION_MEA) then
+        npcUtil.giveKeyItem(player, tpz.ki.MAP_OF_PROMYVION_MEA)
+        player:confirmTrade()
+    else
+        player:messageSpecial(ID.text.NOTHING_HAPPENS)
```

Comment:
Does this text display when you trade a bad item? Bad trades are usually just ignored.

Reply:
"Nothing happens..." is displayed on a bad trade. To my knowledge, that's correct.